xtra-libp2p-rollover/src/v_2_0_0/maker.rs

```diff
@@ -174,6 +174,12 @@ where
             tracing::debug_span!("next rollover message")
         }
 
+        let contract_symbol = self
```

Comment:
I could not make it work  because I think I'd need to spin up a `task` inside the other executor because `get_rates` is async .

Reply:
Wrapping it in `BaseDlcParams` sounds like a different hack to me. This struct is concerned about `Parameters associated with the base DLC involved in a rollover`. I can do that but I think this goes into the wrong direction and puts things together which do not really belong together.

Even though we are _misusing_ the executor to extract something from the CFD I think this is the cleanest of all ugly solutions. 

@luckysori : proposed to create a follow-up ticket which tackles this in a different way.

Reply:
I've added one more commit which decouples the models from the network layer types. This allows us to easier to for #2459

Reply:
> Wrapping it in BaseDlcParams sounds like a different hack to me. This struct is concerned about Parameters associated with the base DLC involved in a rollover. I can do that but I think this goes into the wrong direction and puts things together which do not really belong together.

you can also let `model::Cfd::start_rollover_maker` return `(BaseDlcParams, ContractSymbol)` and then I don't think it would be a hack.

Reply:
Great point! I completely missed that.

In general, I think we shouldn't access from the CFD like this, but this is way less annoying. Thank you for taking a look.